src/Inciter/AMR/mesh_adapter.cpp

```diff
@@ -0,0 +1,813 @@
+#include "mesh_adapter.h"
+
+namespace AMR {
+
+    void mesh_adapter_t::init_node_store(coord_type* m_x, coord_type* m_y, coord_type* m_z, size_t* graph_size)
+    {
+        node_store.set_x(m_x);
+        node_store.set_y(m_y);
+        node_store.set_z(m_z);
+        node_store.m_graphsize = graph_size;
+    }
+
+    void mesh_adapter_t::init_with_nodes(coord_type* m_x, coord_type* m_y, coord_type* m_z, size_t* graph_size)
+    {
+        init_node_store(m_x, m_y, m_z, graph_size);
+        //init(); // TODO: This also needs to call init if you want node support to work
+    }
+
+    void mesh_adapter_t::init(std::vector<size_t> m_tetinpoel, size_t num_nodes) {
```

Comment:
Also, what is the reason for passing the vector by value? If it is not modified (and it isn't) I would pass it by const-ref: `const std::vector<size_t>&`. If it is to be really "consumed", i.e., moved from, then the way to pass it as an rvalue reference: `std::vector<size_t>&&`. Both of these avoid copying.